project/BuildPlugin.scala

```diff
@@ -78,25 +78,30 @@ object BuildKeys {
     Keys.test in AssemblyKeys.assembly := {}
   )
 
+  import sbtbuildinfo.BuildInfoKeys
   val benchmarksSettings: Seq[Def.Setting[_]] = List(
     Keys.skip in Keys.publish := true,
+    BuildInfoKeys.buildInfoKeys := Seq[BuildInfoKey](
+      Keys.resourceDirectory in sbt.Test in sbt.LocalProject("frontend")),
+    BuildInfoKeys.buildInfoPackage := "bloop.benchmarks",
     Keys.javaOptions ++= {
       def refOf(version: String) = {
-        val HasSha = """.*(?:bin|pre)-([0-9a-f]{7,})(?:-.*)?""".r
+        val HasSha = """([0-9a-f]{8})(?:\+\d{8}-\d{4})?""".r
         version match {
           case HasSha(sha) => sha
-          case _ => "v" + version
+          case _ => version
         }
       }
       List(
-        "-DscalaVersion=" + Keys.scalaVersion.value,
-        "-DscalaRef=" + refOf(Keys.scalaVersion.value),
         "-Dsbt.launcher=" + (sys
           .props("java.class.path")
           .split(java.io.File.pathSeparatorChar)
           .find(_.contains("sbt-launch"))
           .getOrElse("")),
-        "-Dbloop.jar=" + AssemblyKeys.assembly.in(sbt.LocalProject("frontend")).value
+        "-DbloopVersion=" + Keys.version.in(sbt.LocalProject("frontend")).value,
```

Comment:
Can we hoist up the local project to another place, reusing the same project reference? Also, can we have the id `"frontend"` be reused both by the `BuildPlugin` and `build.sbt`?